Extension handlers (#18)
jormaechea/open-api-mocker: README.md

paths:
  /cat:
    get:
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  firstName:
                    type: string
                    x-faker: name.firstName
                  lastName:
                    type: string
                    x-faker: name.lastName
                  fullName:
                    type: string
                    x-faker: '{{name.firstName}} {{name.lastName}}'
                  age:
                    type: string
                    x-faker: 'random.number({ "min": 1, "max": 20 })'
                  
``` 

A JSON response similar to the following would be produced:
```JSON
{
    "firstName": "Ted",
    "lastName": "Kozey",
    "fullName": "Herbert Lowe",
    "age": 12
}
```

The _x-faker_ extension accepts values in 3 forms:
1. _fakerNamespace.method_. e.g. `random.uuid`
2. _fakerNamespace.method({ "methodArgs": "in", "json": "format" })_. e.g. `random.number({ "max": 100 })`
3. A mustache template string making use of the 2 forms above. e.g. `My name is {{name.firstName}} {{name.lastName}}`

> *NOTE*: To avoid new fake data from being generated on every call, up to 10 responses per endpoint is cached 
> that is based on the incoming query string, request body and headers.

Reviewer: ```suggestion
*NOTE*: To avoid new fake data from being generated on every call, up to 10 responses per endpoint are cached 
based on the incoming query string, request body and headers.
```

Author: This is what I get for rushing to commit these changes at half 1 in the morning... apologies.

### x-count

> The _x-count_ extension only as affect when used on a `array` type property.

Reviewer: ```suggestion
The _x-count_ extension has effect only when used on an `array` type property.
```
